storage/tokudb/ha_tokudb.cc

```diff
@@ -360,20 +360,40 @@ void TOKUDB_SHARE::set_cardinality_counts_in_table(TABLE* table) {
                 (i == table->s->primary_key) ||
                 (table->key_info[i].flags & HA_NOSAME);
 
-            uint32_t num_key_parts = get_key_parts(&table->key_info[i]);
-            for (uint32_t j = 0; j < num_key_parts; j++) {
+            KEY* key = &table->key_info[i];
+
+            /* Check if this index supports index statistics. */
+            if (!key->supports_records_per_key()) {
+                continue;
+            }
+
+            for (uint32_t j = 0; j < key->actual_key_parts; j++) {
+                if (j >= key->user_defined_key_parts) {
+                    key->set_records_per_key(j, 1.0);
```

Comment:
For this particular check, I think it is fine. Judging by the comments in both the optimizer code and in ha_innodb::info_low this new use of a float is experimental and at some point the old ulong will go away along with (hopefully) some cleanup to make it clear exactly what an SEs requirements are w.r.t. reporting cardinality. So yeah, making these changes doesn't end this, we'll be back in here sooner or later.